### PR TITLE
Hand tracking

### PR DESCRIPTION
**Pull request by superduperpacman42**
Overdue pull request containing the following functionality:
- Pointing gesture detection (verified on drone for vision targets on the floor, but not actual human player)
- Planner node with multi-drone support (verified on drones)
- Voice commands  (verified on drones)
- Obstacle avoidance (only tested in simulator, not integrated with the rest of the ROS pipeline yet)

### CODE REVIEW

File: iarc_ui/src/iarc_ui/whistle.py [hunk 1]
```diff
@@ -73,13 +73,13 @@ def groupToCmd(self, group):
     # Each command has three parts: drone name, direction, distance.
     # These convert the four whistled tones into each of those parts.
     drones = ["alpha", "bravo", "charlie", "delta"]
-    dirs = ["north", "east", "south", "west"]
+    commands = ["north", "east", "south", "west", "west"]
```
Comment (nathanestill) [inline iarc_ui/src/iarc_ui/whistle.py]: Why two wests?

File: iarc_main/launch/backbone.launch [hunk 2]
```diff
@@ -2,6 +2,10 @@
     <node name="arbiter" ns="bebop" pkg="iarc_arbiter" type="arbiter_lite.py" output="screen">
         <remap from="/odometry" to="/bebop/odom"/>
     </node>
+<<<<<<< HEAD
```
Comment (nathanestill) [inline iarc_main/launch/backbone.launch]: Is this supposed to be here?

File: iarc_forebrain/scripts/Planner/voice_interface.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        self.pub = rospy.Publisher("/voice", String, queue_size=10)
+        self.drone = 'alexa'
+
+    def run(self):
+        while not rospy.is_shutdown():
+            output = ''
+            words = listen()
+            if len(words) == 0 or 'cancel' in words:
+                continue
+            if 'shutdown' in words:
```
Comment (HALtheWise) [inline iarc_forebrain/scripts/Planner/voice_interface.py]: ???

File: iarc_main/launch/backbone.launch [hunk 4]
```diff
@@ -2,6 +2,10 @@
...
+
+=======
+>>>>>>> master
```
Comment (HALtheWise) [inline iarc_main/launch/backbone.launch]: Wrong

File: iarc_ui/src/iarc_ui/whistle.py (hunk 1 above)
Comment (HALtheWise) [inline iarc_ui/src/iarc_ui/whistle.py]: ?

File: requirements.txt [hunk 5]
```diff
@@ -1,3 +1,6 @@
 qrcode==6.0
-numpy==1.11.0
+numpy
 matplotlib==1.5.1
+SpeechRecognition
+pocketsphinx
+pyaudio>=0.2.11
```
Comment (HALtheWise) [inline requirements.txt]: TBD, figure out Travis

File: iarc_arbiter/src/iarc_arbiter/drone.py [hunk 6]
```diff
@@ -373,3 +379,9 @@ def move_camera(self, pitch, yaw):
         cameraCoordinates.angular.y = pitch
         cameraCoordinates.angular.z = yaw
         self.gimbalPub.publish(cameraCoordinates)
+
+    def prefix(self, frame):
```
[the remark this answers is not in the document]
Comment (superduperpacman42) [reply iarc_arbiter/src/iarc_arbiter/drone.py]: I think we decided that users of the drone object should just use this function for their prefixing needs

File: iarc_forebrain/scripts/Planner/voice_interface.py (hunk 3 above)
Comment (superduperpacman42) [reply iarc_forebrain/scripts/Planner/voice_interface.py]: Shutdown now lands all drones before terminating the voice interface

File: iarc_forebrain/scripts/Planner/pointing_detection.py [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+angles are measured counterclockwise from positive x (right) axis
+Direction goes from red (helmet) to blue (hand)
+
+'''
+
+PLAYER_HEIGHT = 1.5 # Height of human player color targets off the ground in meters
+FOCAL_X = 520.6 #691 # Camera focal length in pixels
+FOCAL_Y = 514 #689 # Camera focal length in pixels
+IMAGE_WIDTH = 856
+IMAGE_HEIGHT = 480
```
[the remark this answers is not in the document]
Comment (superduperpacman42) [reply iarc_forebrain/scripts/Planner/pointing_detection.py]: These are now autodetected

File: requirements.txt (hunk 5 above)
Comment (superduperpacman42) [reply requirements.txt]: Travis was missing portaudio19-dev, build is now successful

**Pull request merged**